storefront/app/controllers/spree/line_items_controller.rb

```diff
@@ -45,9 +45,12 @@ def update
       quantity = line_item_params[:quantity]&.to_i || 1
       result = cart_set_item_quantity_service.call(order: @order, line_item: @line_item, quantity: quantity)
 
-      @error = result.value.errors.full_messages.to_sentence if result.failure?
+      if result.failure?
+        @error = result.value.errors.full_messages.to_sentence
+        flash.now[:error] = @error
+      end
 
-      load_line_items if result.success?
+      load_line_items
```

Comment:
We need to load line items for `update` and `destroy`, so it doesn't break the cart. It will load current items w/o any changes.